src/components/attach/AttachHub.js

```diff
@@ -0,0 +1,348 @@
+      setIsPending(true);
+
+      inputTextRef.current.value = "";
+      const cid = selectedCID;
+      const mid = currentUser._id + Date.now();
+
+      let msg = {
+        _id: mid,
+        body,
+        from: currentUser._id,
+        t: Date.now(),
+        attachments: files.map((file) => ({
+          file_id: file.name,
+          file_name: file.name,
+          file_url: file.localUrl,
+        })),
+      };
+
+      dispatch(addMessage(msg));
+      dispatch(updateLastMessageField({ cid, msg }));
+
+      let attachments = [];
+      const reqData = { mid, body, cid };
+
+      if (files?.length) {
+        attachments = await DownloadManager.getFileObjects(files);
+        reqData["attachments"] = attachments.map((obj, i) => ({
+          file_id: obj.file_id,
+          file_name: obj.file_name,
+          file_blur_hash: files[i].blurHash,
+        }));
+      }
+
+      let response;
+      try {
+        response = await api.messageCreate(reqData);
+      } catch (err) {
+        showCustomAlert("The server connection is unavailable.", "warning");
+        dispatch(
+          setLastMessageField({ cid, msg: messages[messages.length - 1] })
+        );
+        return;
+      }
+
+      if (response.mid) {
+        msg = {
+          _id: response.server_mid,
+          body,
+          from: currentUser._id,
+          status: "sent",
+          t: response.t,
+          attachments: attachments.map((obj, i) => ({
+            file_id: obj.file_id,
+            file_name: obj.file_name,
+            file_url: obj.file_url,
+            file_local_url: files[i].localUrl,
+            file_blur_hash: files[i].blurHash,
+          })),
+        };
+
+        dispatch(addMessage(msg));
+        dispatch(
+          updateLastMessageField({ cid, resaveLastMessageId: mid, msg })
+        );
+        dispatch(removeMessage(mid));
+      }
+
+      setIsSendMessageDisable(false);
+      setIsPending(false);
+      closeModal(true);
+    },
+    [
+      closeModal,
+      connectState,
+      currentUser,
+      dispatch,
+      files,
+      isSendMessageDisable,
+      messages,
+      selectedCID,
+    ]
+  );
+
+  const handeOnKeyDown = useCallback(
+    (e) => {
+      if (
+        e.keyCode === KEY_CODES.ENTER &&
+        ((!isMobile && !e.shiftKey) || (isMobile && e.shiftKey))
+      ) {
+        e.preventDefault();
+        sendMessage();
+      }
+    },
+    [isMobile, sendMessage]
+  );
+
+  const handleInput = useCallback(
+    (e) => {
+      if (inputTextRef.current) {
+        const countOfLines = e.target.value.split("\n").length - 1;
+        inputTextRef.current.style.height = `calc(${
+          40 + countOfLines * 20 < 230 ? 40 + countOfLines * 20 : 215
+        }px * var(--base-scale)) `;
+        inputTextRef.current.scrollTop = inputTextRef.current.scrollHeight;
+      }
+    },
+    [inputTextRef]
+  );
+
+  const pickFileClick = useCallback(
+    () => inputFilesRef.current.click(),
+    [inputFilesRef]
```

Comment:
ref will not work for dependencies

Reply:
we can just write it as `const pickFileClick = () => inputFilesRef.current.click()`